dulwich/porcelain.py

```diff
@@ -1753,6 +1755,76 @@ def update_head(repo, target, detached=False, new_branch=None):
             r.refs.set_symbolic_ref(b"HEAD", to_set)
 
 
+def reset_file(repo, file_path: str , target: bytes = b'HEAD'):
+    """
+    reset the file to specific commit or branch
+    Args:
+        repo: dulwich Repo object
+        file_path: file to reset, relative to the repository path
+        target: branch or commit or b'HEAD' to reset
+    """
+    if target in branch_list(repo):
```

Comment:
you probably want just parse_tree() here rather than duplicating the logic (see elsewhere in dulwich/porcelain.py)

Reply:
I have remove other Conditionals except
```
if target in branch_list(repo):
        target = b'refs/heads/' + target
```
I think we should keep this to match branch name

Reply:
If we'd like to support branch names here (and I agree that would be a good idea), then we should fix that in parse_tree rather than here.

Reply:
Update parser_tree in the new commit